### DISCUSSION TITLE
How to add search to the filter list

### DISCUSSION THREAD

**Discussion by SebCorbin** [wagtail/wagtail#12738] [Q&A] [answered]
Having a dynamic construction of Filterset on some custom page types and snippets, we are ending up with a lot of filtering possibilities (around 35 filters possible).

Hence, the filters drilldown menu can get quite long, how would you add a searching field (for filter names in this area? I tried overriding `wagtailadmin/shared/headers/_filters.html` but it's tightly coupled to the `SwapController` and the underlying form refreshes the results on input.

**Comment by KianAnbarestani** [comment, marked as answer]
You can add a search field to filter the dropdown without breaking SwapController. Since _filters.html is tightly coupled, the best approach is to inject a simple JavaScript filter instead of overriding core behavior.
